File: day-01/calibrate.py
from functools import reduce
from operator import add

def get_device_input(filename):
    with open(filename) as file:
        values = [int(line.strip()) for line in file]

    return values

if __name__ == '__main__':
    values = get_device_input('input.txt')
    total = reduce(add, values)

    print(total)

File: day-01/calibrate.js
/*
 * Find a final frequency value after incremental changes
 *
 * Problem statement: adventofcode.com/2018/day/1
 */

const fs = require('fs')

// reads integer input values from the input file
const getFrequencyChanges = cb => {
  fs.readFile('input.txt', 'utf8', (err, data) => {
    cb(data.split('\n').map(Number))
  })
}

// read and sum up frequency changes to get the final value
getFrequencyChanges(values => {
  const total = values.reduce((acc, v) => acc + v)
  console.log(total)
})
